rest_api/sawtooth_rest_api/messaging.py

```diff
@@ -161,10 +180,21 @@ def __init__(self, socket, msg_router):
             content=message_content,
             message_type=message_type)
 
-        try:
-            await self._socket.send_multipart([message.SerializeToString()])
-        except asyncio.CancelledError:
-            raise
+        # Send the message. Backoff and retry in case of an error
+        backoff = _Backoff()
+
+        while True:
+            try:
+                await self._socket.send_multipart(
+                    [message.SerializeToString()])
+                break
+            except asyncio.CancelledError:
+                LOGGER.warning('ZMQ connection dropped. Backing off.')
+                try:
+                    await backoff.do_backoff()
+                except DisconnectError:
+                    LOGGER.warning('Retry limit reached for ZMQ. Exiting.')
```

Comment:
This information should be logged on the same line above. Really there should only be one logging statement per backoff event.

This is also misleading, as this condition was not generated by ZMQ itself, though it might be interpreted as such when viewing it in the log.

Reply:
This won't ultimately report the error back to the user, via an HTTP error.